src/IssueCreator/Dialogs/ManageIssue.cs

```diff
@@ -31,52 +26,40 @@ public ManageIssue(IssueManager issueManager, Settings settings, FileLogger logg
             _issueManager = issueManager;
             _logger = logger;
             _settings = settings;
-
-            cboEpics.Items.AddRange(epicList);
+            if(epicList != null)
+            {
+                cboEpics.Items.AddRange(epicList);
+            }
+            
             cboAvailableRepos.Items.AddRange(settings.Repositories.ToArray());
+            if(cboAvailableRepos.Items.Count == 1)
+            {
+                cboAvailableRepos.SelectedIndex = 0;
+            }
         }
 
         private void btnClose_Click(object sender, EventArgs e)
         {
             this.Close();
         }
 
-        private async void btnAdd_Click(object sender, EventArgs e)
-        {
-            using IDisposable scope = _logger.CreateScope("Add issue to epic");
-            await ExecuteIssueOperation(_issueManager.AddIssueToEpicAsync);
-        }
-
-        private async void btnRemove_Click(object sender, EventArgs e)
-        {
-            using IDisposable scope = _logger.CreateScope("Remove issue to epic");
-            await ExecuteIssueOperation(_issueManager.RemoveIssueFromEpicAsync);
-        }
-
-        private async Task ExecuteIssueOperation(Func<long, int, long, int, Task<bool>> action)
+        protected async Task ExecuteIssueOperation(Func<long, int, long, int, Task<bool>> action, string scopeText)
         {
-            if (!int.TryParse(txtIssueNumber.Text, out int issueNumber))
-            {
-                MessageBox.Show("Please specify a number for the issue");
-            }
+            using IDisposable scope = _logger.CreateScope(scopeText);
+            
+            setValidation(true);
 
-            if (cboAvailableRepos.SelectedItem == null)
-            {
-                MessageBox.Show("Please select a repository");
+            if (!this.ValidateChildren())
+            {
+                return;
             }
-
-            if (cboEpics.SelectedItem == null)
-            {
-                MessageBox.Show("Please select an epic");
-            }
-
             // get the repository
             (string owner, string repo) = UIHelpers.GetRepoOwner(cboAvailableRepos.SelectedItem);
 
             RepositoryInfo repoFromGH = await _issueManager.GetRepositoryAsync(owner, repo);
             IssueDescription epicInfo = cboEpics.SelectedItem as IssueDescription;
 
-            bool result = await action(epicInfo.Repo.Id, epicInfo.Issue.Number, repoFromGH.Id, issueNumber);
+            bool result = await action(epicInfo.Repo.Id, epicInfo.Issue.Number, repoFromGH.Id, int.Parse(txtIssueNumber.Text));
```

Comment:
Would it make sense to parse the number above using `TryParse` and then error out or not doing this if the number is not valid?

Reply:
This is now ensured via field validation. So if we got here, it means the inputs have been validated and parsed. It may make sense to store the value as a local `int` property after being validated so we aren't using the `Text` property as the model anymore.

Reply:
see https://github.com/AlexGhiondea/IssueCreator/pull/21/files#r455276654